ui-library-filesharing-ui-components/app/src/ChatComponents.tsx

```diff
@@ -0,0 +1,137 @@
+import {
+  usePropsFor,
+  MessageThread,
+  SendBox,
+  ActiveFileUpload,
+  FileMetadata,
+} from "@azure/communication-react";
+import React from "react";
+import axios from "axios";
+import Form from "form-data";
+import { v4 } from "uuid";
+
+function ChatComponents(): JSX.Element {
```

Comment:
This is a complex component. I'd recommend adding a bunch of explanatory comments inline.

- With each state / ref declaration, note what the variable is tracking, when it is updated, how it relates to other state variables.
- With each inline function declaration, document when it is called.

Given the complexity here, being a bit vebose / repetitive is worthwhile if it helps a new reader understand what is going on.